sdks/blue-krill/blue_krill/encrypt/handler.py

```diff
@@ -50,3 +52,55 @@ def strip_header(cls, text: str):
         @classmethod
         def contain_header(cls, text: str) -> bool:
             return text.startswith(cls.HEADER)
+
+
+# 国密算法
+class NationEncryptHandler:
+    def __init__(self):
+        symmetric_cipher: BaseSymmetricCipher = symmetric_cipher_manager.cipher(using="default")
+        self.f = symmetric_cipher
+
+    def encrypt(self, text: str) -> str:
+        if self.Header.contain_header(text):
+            return text
+
+        return self.Header.add_header(self.f.encrypt(text))
+
+    def decrypt(self, encrypted: str) -> str:
+        encrypted = self.Header.strip_header(encrypted)
+
+        return self.f.decrypt(encrypted)
+
+    class Header:
+        HEADER = "nationcrypto$"
+
+        @classmethod
+        def add_header(cls, text: str):
+            return cls.HEADER + text
+
+        @classmethod
+        def strip_header(cls, text: str):
+            # 兼容无 header 加密串
+            if not cls.contain_header(text):
+                return text
+
+            return text[len(cls.HEADER) :]
+
+        @classmethod
+        def contain_header(cls, text: str) -> bool:
+            return text.startswith(cls.HEADER)
+
+
+# 根据 django setting BKKRILL_ENCRYPT_HANDLER 字段选择相应的加密算法
+# 配置字段为"NationEncryptHandler"时使用国密算法，其余情况均默认使用国际加密算法
+def get_encrypt_handler():
+    try:
+        from django.conf import settings
+
+        cipher_name = settings.BKKRILL_ENCRYPT_HANDLER
+        if cipher_name == "NationEncryptHandler":
+            return NationEncryptHandler()
+        else:
+            return EncryptHandler()
+    except (ImportError, AttributeError):
```

Comment:
except 不应该返回 EncryptHandler
这样很容易出现预期外的问题，尤其这玩意是要存 DB 里的

Reply:
挂了就是挂了，让它挂，只要错误信息明确，改一行配置就能恢复服务，就让它挂。

```
    try:
        from django.core.management import execute_from_command_line
    except ImportError:
        # The above import may fail for some other reason. Ensure that the
        # issue is really that Django is missing to avoid masking other
        # exceptions on Python 2.
        try:
            import django  # noqa
        except ImportError:
            raise ImportError(
                "Couldn't import Django. Are you sure it's installed and "
                "available on your PYTHONPATH environment variable? Did you "
                "forget to activate a virtual environment?"
            )
        raise
```
这是 django 项目的 manage.py 的信息